[Bug]: Data node ListSuccessorMarkets API returns 0 markets

### Problem encountered

The new successor market API no longer returns any markets.
https://jenkins.ops.vega.xyz/job/common/job/system-tests-nightly/999/testReport/

### Observed behaviour

No markets returned by API

### Expected behaviour

All markets in the succession line of the provided market are returned.

### Steps to reproduce

```bash
1. pytest -k test_list_successor_markets_parent_only
```


### Software version

3dc4ccd0158c11f7a39c8ee93c77e260cf2b8986

### Failing test

tests.governance.test_markets.test_list_successor_markets_parent_only

### Jenkins run

https://jenkins.ops.vega.xyz/job/common/job/system-tests-nightly/999/testReport/

### Configuration used

_No response_

### Relevant log output

```shell
context = {'AssetID': '690b27da6c9e6e67ba67f52c4bf753031ea92be8990a32ee3f4b0c13d36c6f4b', 'app_state.network_parameters': {'bloc...de18b8eae987049a4a07a660789f66610c5f29d76ae854e21be7c26b70135935', '0x55Bb84D3FA3A1D8Bd86643F66eaCdfde812fAce5')], ...}
proposer = '8B1BE8F32A'

    @pytest.mark.successor_markets
    @pytest.mark.markets
    @pytest.mark.full
    def test_list_successor_markets_parent_only(context, proposer):
        """
        ListSuccessorMarkets should return the first market in the succession line
        when there are no child markets.
        """
        parent = Market(context, proposer)
        parent.propose(proposer)
    
        markets = marketsTrading.list_successor_markets(parent.id, True)["successorMarkets"]["edges"]
>       assert len(markets) == 1
E       assert 0 == 1
E        +  where 0 = len([])

tests/governance/test_markets.py:551: AssertionError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@
 - [8721](https://github.com/vegaprotocol/vega/issues/8721) - Fix panic with triggered OCO expiring
 - [8729](https://github.com/vegaprotocol/vega/issues/8729) - Stop order direction not set in datanode
 - [8545](https://github.com/vegaprotocol/vega/issues/8545) - Block Explorer pagination does not order correctly.
+- [8748](https://github.com/vegaprotocol/vega/issues/8748) - `ListSuccessorMarkets` does not return results.
 
 
 ## 0.72.1

--- a/datanode/sqlstore/markets.go
+++ b/datanode/sqlstore/markets.go
@@ -340,5 +340,20 @@ left join lineage s on l.successor_market_id = s.parent_id
 		edges = append(edges, edge)
 	}
 
+	if len(markets) == 0 {
+		// We do not have any markets in the given succession line, so we need to return the market
+		// associated with the given market ID, which should be the parent market.
+		market, err := m.GetByID(ctx, marketID)
+		if err != nil {
+			return nil, entities.PageInfo{}, err
+		}
+
+		edge := entities.SuccessorMarket{
+			Market: market,
+		}
+
+		edges = append(edges, edge)
+	}
+
 	return edges, pageInfo, nil
 }

--- a/datanode/sqlstore/markets_test.go
+++ b/datanode/sqlstore/markets_test.go
@@ -1173,6 +1173,17 @@ func testListSuccessorMarkets(t *testing.T) {
 			EndCursor:       wantEndCursor,
 		}, pageInfo)
 	})
+
+	t.Run("should list the parent market even if it has not entered continuous trading and has no successors", func(t *testing.T) {
+		got, _, err := md.ListSuccessorMarkets(ctx, "deadbeef04", false, entities.CursorPagination{})
+		require.NoError(t, err)
+		want := []entities.SuccessorMarket{
+			{
+				Market: markets[10],
+			},
+		}
+		assert.Equal(t, want, got)
+	})
 }
 
 func testGetMarketWithParentAndSuccessor(t *testing.T) {
@@ -1236,6 +1247,14 @@ func setupSuccessorMarkets(t *testing.T, ctx context.Context) (*sqlstore.Markets
 			TradableInstrument: &vega.TradableInstrument{},
 		},
 		ParentMarketID: successorMarketA.ID,
+	}
+
+	parentMarket2 := entities.Market{
+		ID:           entities.MarketID("deadbeef04"),
+		InstrumentID: "deadbeef04",
+		TradableInstrument: entities.TradableInstrument{
+			TradableInstrument: &vega.TradableInstrument{},
+		},
 	}
 
 	successorMarketA.SuccessorMarketID = successorMarketB.ID
@@ -1399,6 +1418,11 @@ func setupSuccessorMarkets(t *testing.T, ctx context.Context) (*sqlstore.Markets
 			market:      successorMarketB,
 			state:       entities.MarketStateActive,
 			tradingMode: entities.MarketTradingModeContinuous,
+		},
+		{
+			market:      parentMarket2,
+			state:       entities.MarketStatePending,
+			tradingMode: entities.MarketTradingModeOpeningAuction,
 		},
 	}
 

